### DISCUSSION TITLE
Adopt a style guide for docstrings

### DISCUSSION THREAD

**Discussion by EwoutH** [mesa/mesa#1499] [General]
It might be useful to adopt a style guide for docstrings. Common ones:

- Strict [PEP 257 Docstring Conventions](https://peps.python.org/pep-0257/)
- The [Google style guide](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings)
- The [NumPy style guide](https://numpydoc.readthedocs.io/en/latest/format.html)
- The [Sphinx docstring format](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html)

For example, the NumPy style guide maintains a white line before the last `"""`, while the others do not.

**Comment by rht** [comment]
It looks like this can be linted with https://github.com/PyCQA/pydocstyle.

**Reply by rht** [reply]
Auto-formatter: https://github.com/PyCQA/docformatter

**Comment by rht** [comment]
My natural preference is without the extra line, but I'd want to know why NumPy, SciPy, and the scikits choose to have it. Their guideline doesn't say anything about it.

**Comment by Corvince** [comment]
Good idea. I would naturally prefer the nummy style guide, since we are part of the scientific python ecosystem.

The thing with the empty line before the triple quotes seems useless to me but shouldn't be a decisive factor imho, it is rather pointless to discuss about such a matter

**Reply by rht** [reply]
[pandas](https://github.com/pandas-dev/pandas/blob/a793802ca08ba159558e36db95f8242fb0f44156/pandas/core/algorithms.py#L135) and [NetworkX](https://github.com/networkx/networkx/blob/6ef8b9986ad9a8bc79a4a6640a8f9ee285b67a7b/networkx/algorithms/asteroidal.py#L61) don't have the extra line. Surprisingly, [scikit-learn](https://github.com/scikit-learn/scikit-learn/blob/80d21c37672dbb4b2439fbfb19b8d5e28d7f20ab/sklearn/compose/_target.py#L128) doesn't have it either.

**Reply by Corvince** [reply]
Also consider that guide lines are guide lines, not strict rules. The blank line doesn't seem important and there is no reason not to follow the numpy guide lines, but skip the last line. (as seen in linked packages by @rht )
